### PR TITLE
Introduce configuration to disable OpenAPI doc autogeneration

### PR DESCRIPTION
**Pull request by ayeshLK**
## Purpose
With the latest OpenAPI built-in compiler plugin, for every ballerina HTTP service there will be an OpenAPI doc generated and it will be hosted in an introspection endpoint for that service. But, the developer should have the control to disable this feature if needed.

Fixes [#1921](https://github.com/ballerina-platform/ballerina-standard-library/issues/1921)

## Samples
Consider following scenario and expected behavior from OpenAPI built-in plugin.

### `embed` Configuration
```ballerina
@openapi:ServiceInfo {
    embed: false
}
@http:ServiceConfig {compression: {enable: http:COMPRESSION_AUTO}}
service / on new http:Listener(9090) {
    resource function get greeting() returns string {
        return "Hello, World!";
    }

    // implement other resource methods
}
```
If above configurations are available generated resources will not be added  to the executable JAR.

### CODE REVIEW

File: module-ballerina-openapi/annotation.bal [hunk 1]
```diff
@@ -21,13 +21,17 @@
...
+# + autoEmbedToService - Enable auto-inject of OpenAPI documentation to current service
+# + autoGen - Enable OpenAPI documentation generation for current service
 public type ServiceInformation record {|
     string contract = "";
     string[]? tags = [];
     string[]? operations = [];
     string[]? excludeTags = [];
     string[]? excludeOperations = [];
     boolean failOnErrors = true;
+    boolean autoEmbedToService = true;
```
Comment (hevayo) [inline module-ballerina-openapi/annotation.bal:33]: Can this be `embedToService` or simply `embed`

**Review by hevayo — COMMENTED** [review]
please see inline comment

**Comment by shafreenAnfar** [comment]
Please note that we only need `autoEmbedToService`. Setting `false` to the configuration implicitly means not to generate as well.

File: module-ballerina-openapi/annotation.bal (hunk 1 above)
Comment (ayeshLK) [reply module-ballerina-openapi/annotation.bal]: Addressed in [712ae126126dae27896b5f1eed5f0e0a063c2a07](https://github.com/ballerina-platform/ballerina-openapi/pull/599/commits/712ae126126dae27896b5f1eed5f0e0a063c2a07).

**Review by lnash94 — APPROVED** [review]

**Pull request merged**